python/cog/server/runner.py

```diff
@@ -51,13 +51,21 @@ def setup(self) -> None:
         self.predictor_process = multiprocessing.get_context("spawn").Process(
             target=self._start_predictor_process
         )
+
+        self._is_processing = True
         self.predictor_process.start()
 
+        while self.is_processing():
```

Comment:
Good shout. I've added an infinite poll here. We might need to add some more logic to prevent failed setups from tying up the main process forever, but that's a problem with the current behaviour anyway I think 🤔